doc/man7/EVP_PKEY-DH.pod

```diff
@@ -175,8 +174,51 @@ An B<DH> key can be generated using domain parameters by calling:
     EVP_PKEY_free(key);
     EVP_PKEY_CTX_free(gctx);
 
-=for comment TODO(3.0): To validate domain parameters, additional values used
-during generation may be required to be set into the key.
+To validate B<FIPS 186-4> B<DHX> domain parameters decoded from B<PEM> or
+B<DER> data, additional values used during generation may be required to
+be set into the key.
+
+EVP_PKEY_todata(), OSSL_PARAM_merge(), and EVP_PKEY_fromdata() are useful
+to add these parameters to the original key or domain parameters before
+the actual validation.
+
+    EVP_PKEY *received_domp = ...; /* parameters received and decoded */
+    unsigned char *seed = ...;     /* and additional parameters received */
+    size_t seedlen = ...;          /* by other means, required */
+    int gindex = ...;              /* for the validation */
+    int pcounter = ...;
+    int hindex = ...;
+    OSSL_PARAM extra_params[5];
+    OSSL_PARAM *domain_params = NULL;
+    OSSL_PARAM *merged_params = NULL;
+    EVP_PKEY_CTX *ctx = NULL, *validate_ctx = NULL;
+    EVP_PKEY *complete_domp = NULL;
+
+    EVP_PKEY_todata(received_domp, OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS,
+                    &domain_params);
+    extra_params[0] = OSSL_PARAM_construct_octet_string("seed", seed, seedlen);
+    extra_params[1] = OSSL_PARAM_construct_int("gindex", &gindex);
+    extra_params[2] = OSSL_PARAM_construct_int("pcounter", &pcounter);
+    extra_params[3] = OSSL_PARAM_construct_int("hindex", &hindex);
+    extra_params[4] = OSSL_PARAM_construct_end();
+    merged_params = OSSL_PARAM_merge(domain_params, extra_params);
+
+    ctx = EVP_PKEY_CTX_new_from_name(NULL, "DHX", NULL);
+    EVP_PKEY_fromdata_init(ctx);
```

Comment:
It is probably not great that none of these check the return values (see todata above also)

Reply:
We do not usually add error checking to the examples in man pages as it makes the example code hard to read. But perhaps a code comment adding the example does not do proper error checking should be added.